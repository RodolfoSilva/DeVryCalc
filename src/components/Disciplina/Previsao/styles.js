import { StyleSheet } from 'react-native'
import { colors } from '../../../styles'

export default StyleSheet.create({
  container: {
    paddingBottom: 10
  },
  value: {
    fontSize: 50,
    paddingVertical: 6
  }
})
