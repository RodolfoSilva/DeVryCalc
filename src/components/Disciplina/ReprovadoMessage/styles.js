import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingVertical: 26
  },
  message: {
    fontSize: 14,
    fontWeight: '500',
    color: '#333'
  }
})
