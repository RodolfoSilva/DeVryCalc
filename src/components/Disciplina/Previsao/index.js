import React from 'react'
import PropTypes from 'prop-types'
import Nota from '../Nota'
import styles from './styles'

const Previsao = ({ value, title }) => (
  <Nota
    value={value}
    title={title}
    style={styles.container}
    valueStyle={styles.value}
  />
)

Previsao.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string.isRequired
}
export default Previsao
