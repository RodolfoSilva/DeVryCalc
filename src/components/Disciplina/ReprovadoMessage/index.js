import React from 'react'
import { Text, View } from 'react-native'
import styles from './styles'

const ReprovacaoMessage = () => (
  <View style={styles.container}>
    <Text style={styles.message}>
      Não foi desssa vez, sorte na próxima....
    </Text>
  </View>
)

export default ReprovacaoMessage
