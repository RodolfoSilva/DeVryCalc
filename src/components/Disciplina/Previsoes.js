import React from 'react'
import { Text, View } from 'react-native'
import Previsao from './Previsao'
import { NotasCalculator } from '../../utils'

const Previsoes = ({ notas }) => {
  const mediaFinal = NotasCalculator.mediaFinal(notas)
  const notaParaAP1 = NotasCalculator.notaParaAP1(notas)
  const notaParaAP2 = NotasCalculator.notaParaAP2(notas)
  const notaParaAP3 = NotasCalculator.notaParaAP3(notas)

  return (
    <View>
      {mediaFinal !== null && <Previsao value={mediaFinal} title="Média final" />}
      {notaParaAP1 !== null && notaParaAP1 <= 10 && <Previsao value={notaParaAP1} title="Necessário tirar na AP1" />}
      {notaParaAP2 !== null && notaParaAP2 <= 10 && <Previsao value={notaParaAP2} title="Necessário tirar na AP2" />}
      {notaParaAP3 !== null && notaParaAP3 <= 10 && <Previsao value={notaParaAP3} title="Necessário tirar na AP3" />}
      {notaParaAP3 > 10 && <View style={{ alignItems: 'center', paddingVertical: 26 }}>
        <Text style={{ fontSize: 14, fontWeight: '500', color: '#333' }}>Não foi desssa vez, sorte na próxima....</Text>
      </View>}
    </View>
  )
}


export default Previsoes
