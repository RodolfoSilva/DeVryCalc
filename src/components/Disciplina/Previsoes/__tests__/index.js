import React from 'react'
import Previsoes from '../'
import Previsao from '../../Previsao'
import ReprovadoMessage from '../../ReprovadoMessage'
import { shallow } from 'enzyme'

describe('Previsoes', () => {
  it('Renderiza somente a view se todas as notas estivereme em branco', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: null, ap2: null, ap3: null }} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a média final', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: 0, ap2: 0, ap3: 0 }} />)
    expect(wrapper.find(Previsao).first().prop('title')).toEqual('Média final')
    expect(wrapper.find(Previsao).first().prop('value')).toEqual(0)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a mensagem de reprovação', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: 0, ap2: 0, ap3: null }} />)
    expect(wrapper.find(Previsao).length).toEqual(0)
    expect(wrapper.find(ReprovadoMessage).length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a previsão para AP1', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: null, ap2: 5, ap3: 5 }} />)
    expect(wrapper.find(Previsao).first().prop('title')).toEqual('Necessário tirar na AP1')
    expect(wrapper.find(Previsao).first().prop('value')).toEqual(5)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a previsão para AP2', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: 5, ap2: null, ap3: 5 }} />)
    expect(wrapper.find(Previsao).first().prop('title')).toEqual('Necessário tirar na AP2')
    expect(wrapper.find(Previsao).first().prop('value')).toEqual(5)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a previsão para AP3', () => {
    const wrapper = shallow(<Previsoes notas={{ ap1: 5, ap2: 5, ap3: null }} />)
    expect(wrapper.find(Previsao).first().prop('title')).toEqual('Necessário tirar na AP3')
    expect(wrapper.find(Previsao).first().prop('value')).toEqual(5)
    expect(wrapper).toMatchSnapshot()
  })
})
