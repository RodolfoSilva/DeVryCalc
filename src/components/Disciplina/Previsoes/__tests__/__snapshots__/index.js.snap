// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previsoes Renderiza a mensagem de reprovação 1`] = `
<View>
  <ReprovadoMessage />
</View>
`;

exports[`Previsoes Renderiza a média final 1`] = `
<View>
  <Previsao
    title="Média final"
    value={0}
  />
</View>
`;

exports[`Previsoes Renderiza a previsão para AP1 1`] = `
<View>
  <Previsao
    title="Necessário tirar na AP1"
    value={5}
  />
</View>
`;

exports[`Previsoes Renderiza a previsão para AP2 1`] = `
<View>
  <Previsao
    title="Necessário tirar na AP2"
    value={5}
  />
</View>
`;

exports[`Previsoes Renderiza a previsão para AP3 1`] = `
<View>
  <Previsao
    title="Necessário tirar na AP3"
    value={5}
  />
</View>
`;

exports[`Previsoes Renderiza somente a view se todas as notas estivereme em branco 1`] = `<View />`;
